orca-peering/src/main/kotlin/com/netflix/spinnaker/orca/peering/PeeringAgent.kt

```diff
@@ -156,43 +160,49 @@ class PeeringAgent(
    * There is no harm (just some wasted RDS CPU) to "deleting" an execution that doesn't exist
    */
   private fun peerDeletedExecutions() {
-    val deletedExecutionIds = srcDB.getDeletedExecutions(deletedExecutionCursor)
-    val orchestrationIdsToDelete = deletedExecutionIds.filter { it.execution_type == ExecutionType.ORCHESTRATION.toString() }.map { it.execution_id }
-    val pipelineIdsToDelete = deletedExecutionIds.filter { it.execution_type == ExecutionType.PIPELINE.toString() }.map { it.execution_id }
+    try {
+      val deletedExecutionIds = srcDB.getDeletedExecutions(deletedExecutionCursor)
+      val orchestrationIdsToDelete = deletedExecutionIds.filter { it.execution_type == ExecutionType.ORCHESTRATION.toString() }.map { it.execution_id }
+      val pipelineIdsToDelete = deletedExecutionIds.filter { it.execution_type == ExecutionType.PIPELINE.toString() }.map { it.execution_id }
+
+      log.debug("Found ${deletedExecutionIds.size} (orchestrations: ${orchestrationIdsToDelete.size} pipelines: ${pipelineIdsToDelete.size} deleted candidates after cursor: $deletedExecutionCursor")
+      var hadFailures = false
+      var orchestrationsDeleted = 0
+      var pipelinesDeleted = 0
+
+      try {
+        orchestrationsDeleted = destDB.deleteExecutions(ExecutionType.ORCHESTRATION, orchestrationIdsToDelete)
+        peeringMetrics.incrementNumDeleted(ExecutionType.ORCHESTRATION, orchestrationsDeleted)
+      } catch (e: Exception) {
+        log.error("Failed to delete some orchestrations", e)
+        peeringMetrics.incrementNumErrors(ExecutionType.ORCHESTRATION)
+        hadFailures = true
+      }
 
-    log.debug("Found ${deletedExecutionIds.size} (orchestrations: ${orchestrationIdsToDelete.size} pipelines: ${pipelineIdsToDelete.size} deleted candidates after cursor: $deletedExecutionCursor")
-    var hadFailures = false
-    var orchestrationsDeleted = 0
-    var pipelinesDeleted = 0
+      try {
```

Comment:
I just noticed the code duplication here. I would suggest the following:

  data class DeletionResult(val numDeleted: Int, val hadFailures: Boolean)

  private fun delete(executionType: ExecutionType, idsToDelete: List<String>): DeletionResult {
    var numDeleted = 0
    var hadFailures = false
    try {
      numDeleted = destDB.deleteExecutions(executionType, idsToDelete)
      peeringMetrics.incrementNumDeleted(executionType, numDeleted)
    } catch (e: Exception) {
      log.error("Failed to delete some $executionType", e)
      peeringMetrics.incrementNumErrors(executionType)
      hadFailures = true
    }

    return DeletionResult(numDeleted, hadFailures)
  }

And then in `peerDeletedExecutions`, it becomes simply:

    val (orchestrationsDeleted, orchestrationsHadFailures) = delete(ExecutionType.ORCHESTRATION, orchestrationIdsToDelete)
    val (pipelinesDeleted, pipelinesHadFailures) = delete(ExecutionType.PIPELINE, pipelineIdsToDelete)

Reply:
thanks, i like it!